package gbeic.bbsplusplus.ui.morphing;

import mchorse.bbs_mod.ui.forms.IUIFormList;
import mchorse.bbs_mod.ui.framework.UIContext;
import mchorse.bbs_mod.ui.framework.elements.utils.EventPropagation;
import mchorse.bbs_mod.utils.colors.Colors;
import org.lwjgl.glfw.GLFW;

public class UIBBSPPFormEditorList extends UIBBSPPFormList
{
    public UIBBSPPFormEditorList(IUIFormList palette)
    {
        super(palette);

        this.edit.removeFromParent();
        this.eventPropagataion(EventPropagation.BLOCK_INSIDE).markContainer();
    }

    @Override
    public boolean subKeyPressed(UIContext context)
    {
        if (context.isPressed(GLFW.GLFW_KEY_ESCAPE))
        {
            this.palette.exit();
        }

        return true;
    }

    @Override
    public void render(UIContext context)
    {
        this.area.render(context.batcher, Colors.A50);

        super.render(context);
    }
}
